Various Antagonist Handgun/Rifle changes; Stechkin, Stechkin APS, c20-r, m90gl

necromanceranne opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Alters a few of the antagonist weapons, primarily the Soviet handguns.

**Stechkin**
Now a makarov, using the old Stechkin APS sprites with slight tweaks. It's now chambered for 9mm. Comes with the same ammo diversity as before. Same damage as before. As a consequence, all 9mm using guns do 30 damage. Wowzers.

**Stechkin APS**
Now actually a normal sized gun. This gun isn't small. It's about as big as a m1911 and apparently pretty heavy. It's now available in nuke op uplinks, mostly so you don't have to buy a specific bundle just to get it. Nukies could ALREADY buy this gun, it's just in a bundle. Now you can buy it separately. It also shoots fast and sprays bullets everywhere.

![Screenshot 2020-05-04 10 29 13](https://user-images.githubusercontent.com/40847847/80929869-ce3e1a80-8df2-11ea-81aa-49dc40a63600.png)
![Screenshot 2020-05-04 10 26 32](https://user-images.githubusercontent.com/40847847/80929871-d0a07480-8df2-11ea-926f-e82bf35f03d3.png)

The Stechkin APS costs 10 TC. Same as the c20-r before. Because it literally does the same damage that the c20-r used to do. Shock horror. No ammo variety but nothing stopping you buying some 9mm ammo and filling the APS magazines with it.

**10mm Bullets**
Now do pretty sizable amounts of damage. Though the amount of guns that actually USE 10mm is...like, just the surplus rifle, flat guns I think and the old Stechkin. Really goddamn rare bullet type.

Basically does 10 more damage across the board.

**c20-R and most .45 guns**
.45 now does 35 damage. It's only slightly better per bullet than 9mm.

The c20-R now has more ammo options out of the uplink.

c20-R is now 13 TC. It shoots 3 bullets, that's kind of scary.

**m90gl and anything using 5.56mm**

Now has an innate 30 bullet penetration. This shouldn't appreciably change too much, as it was a one burst crit on an unarmored target before.

Added phasic 5.56mm. This ammunition passes through all objects, like penetrators do. It does significantly less damage per bullet. You can buy it for 8 TC.

## Why It's Good For The Game

It's honestly soothing to my OCD knowing that the soviet era guns are a little closer to their real world equivalents. The APS in particular was always a standout as a weird gun compared to it's real life version.

I kinda felt the c20-r was grossly overshadowed lately as far as a gun choices go. Hopefully this makes it more attractive.

## Changelog
:cl:
tweak: The stechkin pistol is now a makarov. It comes chambered in 9mm. Nothing has really changed for performance or ammo availability. It's just all in 9mm now.
tweak: The stechkin APS is now normal sized and available to nukies for 10 tc. It sprays bullets like mad and fires very quickly.
tweak: The caravan now has stechkin APS magazines.
tweak: The Spetznaz bundle uses incendiary magazines for the APS.
balance: .45 is stronger across the board by about 5 damage, and has more bullet variety.
balance: the c20-r is a 3 burst weapon and costs 13tc.
balance: 5.56mm now has 30 armor penetration and has some bullet spread.. This only affects the m90gl and the Boarder assault rifles.
tweak: You can buy phasic ammo for the m90 for 8 tc out of the uplink. It passes through walls and objects, and does significantly less damage. (comparatively speaking of course)
tweak: X-Ray beams now pass through doors, machinery, and structures like lockers.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

necromanceranne committed:
overhauls the crap out of a lot of antag guns and their ammo

necromanceranne committed:
Should probably shoot an additional bullet per squeeze at this point

necromanceranne committed:
nyoops

partyaddict commented:
we're in the year 2500, i believe the syndicate would use something more modern than a 600 year old soviet handgun

then again we have mosins and 1911s so i don't even know how i can make that judgement

necromanceranne commented:
You do realize the stechkin is from the same era as the makarov, right?

necromanceranne committed:
So travis will stop yelling at me for new lines

ATH1909 commented:
Make sure that this won't break any map, away mission, or ruin designs.

For example, the wishgranter away mission has boxes of 10mm ammo scattered throughout it that are intended to be used with the stechkin(s) that you can find in that away mission.

partyaddict commented:
> 
> 
> You do realize the stechkin is from the same era as the makarov, right? They were in service at pretty much the exact same time.

fuck

necromanceranne commented:
Travis is yelling  at me right now for broken maps so gimme a bit.

necromanceranne committed:
Should be all I need to fix for maps

optimumtact commented:
hate how we have so many ammo types

necromanceranne committed:
Touches up the stechkin sprite again

Pugie commented:
While we're at it make stechkin aps fit in holsters...

necromanceranne commented:
Oooh good point.

senjx commented:
i don't like the C-20R or the MGL change but the stetchkin and makarov changes are good but i would like to see a slight fire delay added on the makarov just to not make it a god-tier firearm out of the box

bluetshirtguy commented:
fucking sweet lookin, nice.

necromanceranne commented:
The benefit of the LMG is largely it's magazine. The c-20r has a magazine size of 24, the m-90gl has 30, and the lmg 50. By comparison, the APS has 15, which is about 5 shots before you empty your mag. the smg is 7, the m-90 10, and the lmg about 16?

This can make a difference in the middle of a fight. Being able to just kind of keep clicking and keep shooting as you go forward is a useful ability and worth the price. The other guns end up having to reload several times by the time the LMG has probably killed a good dozen or so people.

I'll be honest, I don't genuinely know what kind of role the m-90gl is meant to fill but innately armor piercing is interesting and does give it some value over the other guns.

necromanceranne commented:
Oh and the APS should be fitting into the standard array of holsters. The only one it DOESN'T is the detective's. I could change that. It actually fits pistol ammo already.

Tlaltecuhtli commented:
nerf l6 saw

necromanceranne commented:
Why?

HomocidlePancakes commented:
> It's honestly soothing to my OCD knowing that the soviet era guns are a little closer to their real world equivalents.

> Makarov shoots 9mm

It shoots 9x18 ROUNDS REEEEEEEEE IT’S DIFFERENT

necromanceranne committed:
Increased the fire rate and spread of the APS and the spread of the m90, added additional magazine types for the APS and added incendiary mags into the Spetznaz bundle to retain the fire theme, added phasic m90 ammo that penetrates through all surfaces but does significantly less damage available to ops for 8 tc, tided up the code for the aps a little more and also it's sprite.

necromanceranne commented:
A LITTLE CLOSER, IT'S NOT IDENTICAL AAAAAH

necromanceranne commented:
Anyway because I loved the idea of spraying the APS everywhere while dual-wielding the guns, I've increased the spread and fire rate of the gun siginificantly. Good luck hitting anything with it but it'll at least be fun. I also made the magazines in the Spetznaz bundle incendiaries to fit the theme.

I've changed 5.556mm to the old damage value but now it is innately armor piercing. The m90 also has a little bit of spread. However, it also comes with a special new ammo type. Phasic bullets. These do very small (comparatively) amounts of damage but can phase through everything.  Machine gun people down through everything and everyone! Ask relevant questions like 'why are my bullets haunted and whispering to me?' and then not get an answer! Never receive a refund even after your entire lineage is cursed by the gun ghosts! Fun for the whole family!

And finally I changed up the sprite on the APS some more and tided the code.

Also I have no idea why travis is angry!

necromanceranne committed:
fixes some stuff

necromanceranne commented:
Wouldn't you know it, penetrator rounds and these new rounds are broken.

Aaaaaaaaaaaaaaah

necromanceranne committed:
So travis will cease screaming

necromanceranne committed:
Changed how the phasic bullets work to being closer to xray beams, added additional flags to allow the bullets to pass through structures and machinery

necromanceranne committed:
Merge branch 'stechkin_redone' of https://github.com/necromanceranne/tgstation into stechkin_redone

Rohesie commented on code/game/machinery/_machinery.dm:
```diff
@@ -414,6 +414,12 @@ Class Procs:
 		updateUsrDialog()
 
+/obj/machinery/CanAllowThrough(atom/movable/mover, turf/target)
+	.= ..()
```
```suggestion
	. = ..()
```

Rohesie commented on code/game/machinery/_machinery.dm:
```diff
@@ -414,6 +414,12 @@ Class Procs:
+/obj/machinery/CanAllowThrough(atom/movable/mover, turf/target)
+	.= ..()
+	
+	if(istype(mover) && (mover.pass_flags & PASSMACHINE))
```
```suggestion
	if(mover.pass_flags & PASSMACHINE)
```
Only movable atoms can move.

Rohesie commented on code/game/machinery/doors/door.dm:
```diff
@@ -146,6 +146,10 @@
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	
+	if(istype(mover) && (mover.pass_flags & PASSMACHINE))
```
```suggestion
	if(mover.pass_flags & PASSMACHINE)
```

Rohesie replied:
Wait, isn't this already done on the parent?

necromanceranne replied:
For some reason due to the pass check already on the door, I had to add it here as well just to make sure it was checking for this flag. Not sure what a better way would be but it does work where it didn't before.

Rohesie commented on code/game/objects/structures.dm:
```diff
@@ -113,6 +113,12 @@
 			. += examine_status
 
+/obj/structure/CanAllowThrough(atom/movable/mover, turf/target)
+	.= ..()
+	
+	if(istype(mover) && (mover.pass_flags & PASSSTRUCTURE))
```
```suggestion
	. = ..()
	
	if(mover.pass_flags & PASSSTRUCTURE)
```

Rohesie commented on code/modules/projectiles/ammunition/ballistic/pistol.dm:
```diff
@@ -33,8 +33,13 @@
+	projectile_type = /obj/projectile/bullet/c9mm_hp
 
-/obj/item/ammo_casing/c9mm/inc
+/obj/item/ammo_casing/c9mm/fire
```
`/obj/item/ammo_casing/c9mm/incendiary` might be better
`fire` is used for a lot of things on the codebase

necromanceranne replied:
It's not really standardized. I could go through and do all that just to tidy it up and make it more readable

Rohesie replied:
That'd be a pretty positive change, though it's not really required.

Rohesie commented on code/modules/projectiles/boxes_magazines/external/pistol.dm:
```diff
@@ -39,17 +21,61 @@
+	max_ammo = 15
+
+/obj/item/ammo_box/magazine/m9mm_aps/update_icon()
+	..()
```
```suggestion
	. = ..()
```

Rohesie commented:
Personally I'd prefer weapons whose design is based on old models if needed, but not outright old models.

necromanceranne committed:
Update code/game/machinery/_machinery.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

necromanceranne committed:
Update code/game/machinery/_machinery.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

necromanceranne committed:
Update code/game/machinery/doors/door.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

necromanceranne committed:
Update code/modules/projectiles/boxes_magazines/external/pistol.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

necromanceranne committed:
Update code/game/objects/structures.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

necromanceranne commented:
Well we had been using the 'stechkin' for some time, and presumably if that is based on an old soviet gun, there isn't much difference between that and a makarov. There didn't seem to be any guns appearance wise similar to our stechkin designed by Stechkin that I could find. It actually looked like a makarov. And the old APS also looked like this probably because the Stechkin APS is very reminiscent in appearance to the Makarov.

What exactly our stechkin was is a mystery to me. Stechkin wasnt a brand. It was the name of its designer. Maybe it was named that because the name was obscure enough to be passed off as something obscure, old and off-market to work in with the theme of Syndicate weapons being outdated by the games standards. As an alternative, I could just restore the old sprite, rename it to something original and restore it to 10mm and retain the new APS. Though I did like the cross ammo exchangeable nature of the APS and the pistols.

Rohesie commented:
Anything is good to me, it's not a hard opinion besides stating a personal preference on the guns matching the setting.

necromanceranne commented:
The APS is already in the uplink next to the mosin. It would seem thr Syndicate have an assortment of old soviet weaponry for whatever reason.

Rohesie commented:
:^)

necromanceranne committed:
updated the gun sprites a bit more

necromanceranne committed:
Merge branch 'master' into stechkin_redone

necromanceranne committed:
Hopefully this fixes the conflicts?

Rohesie commented on code/game/machinery/doors/door.dm:
```diff
@@ -146,6 +146,10 @@
 
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	
+	if(mover.pass_flags & PASSMACHINE)
+		return TRUE
+		
```
Isn't this already handled on the parent? Looks like a repeated check.

Rohesie commented on code/game/machinery/doors/door.dm:
```diff
@@ -146,6 +146,10 @@
 
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	
+	if(mover.pass_flags & PASSMACHINE)
+		return TRUE
```
```suggestion
	if(.)
		return
```

necromanceranne replied:
oh that works? Cool.

necromanceranne committed:
Update code/game/machinery/doors/door.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

Rohesie commented:
Code is fine. Sprites pending approval by Twaticus.

necromanceranne committed:
Merge branch 'master' into stechkin_redone

necromanceranne committed:
Merge branch 'master' into stechkin_redone

necromanceranne committed:
More updating, fatter stechkin sprite
